### DISCUSSION TITLE
No animation in navigation using dispatch({increment: number, duration: number})

### DISCUSSION THREAD

**Discussion by amarnath-cohesive** [igordanchenko/yet-another-react-lightbox#81] [Q&A] [answered]
Hi folks, I am trying to create a custom navigation component with one button to navigate to the previous slide, and another to navigate to the next. I am using the dispatch function provided by the useLightboxState hook. I have tried setting the value of _duration_ to different values but to no avail. It just quickly snaps to the next image without any animation. Am I missing something here?
Here's the sandbox https://codesandbox.io/s/yet-another-react-lightbox-40-forked-jx78oc?file=/src/examples/PhotoGallery.js

Thank you.

**Comment by igordanchenko** [comment, marked as answer]
**v3 UPDATE**

v3 now provides dedicated `prev` / `next` methods to navigate to the previous / next slide. These methods can be accessed in one of the following ways:

- inside the lightbox - via `useController` hook
- outside the lightbox - via [Controller Ref](https://yet-another-react-lightbox.com/documentation#ControllerRef)

---

It is expected behavior. Modifying the lightbox state in such a way skips the first step in FLIP animation. 

Please use `ACTION_PREV` / `ACTION_NEXT` events to navigate to the previous/next slide.

```jsx
const { publish } = useEvents();

publish(ACTION_NEXT);
```

**Reply by amarnath-cohesive** [reply]
Thank you so much for the answer. This library is super awesome. Thank you for maintaining it.
